test/realm_test.dart

```diff
@@ -1182,20 +1201,78 @@ Future<void> main([List<String>? args]) async {
       final dog = Dog('Fido', owner: person);
 
       expect(person, person);
-      expect(person, isNot(1)); 
+      expect(person, isNot(1));
       expect(person, isNot(dog));
 
       realm.write(() {
         realm..add(person)..add(dog);
       });
 
       expect(person, person);
-      expect(person, isNot(1)); 
+      expect(person, isNot(1));
       expect(person, isNot(dog));
 
       final read = realm.query<Person>("name == 'Kasper'");
 
       expect(read, [person]);
     });
   });
+  group('Cycle referenced objects tests:', () {
+    test('Realm adding objects graph', () {
+      var studentMichele = Student(1)
```

Comment:
minor: could we write on one line 
but more importantly aren't these available in the constructor so we could write like below

Reply:
Fixed in https://github.com/realm/realm-dart/pull/196/commits/e058f5ce6ce1ce13fc88117feebaad19dd5896c7
Both ways are shown.